central/reports/service/v2/service_impl.go

```diff
@@ -147,6 +147,17 @@ func (s *serviceImpl) UpdateReportConfiguration(ctx context.Context, request *ap
 		return nil, errors.Wrapf(errox.NotFound, "report configuration with id '%s' does not exist", request.GetId())
 	}
 
+	query := search.NewQueryBuilder().AddExactMatches(search.ReportConfigID, request.GetId()).AddExactMatches(search.ReportState, storage.ReportStatus_WAITING.String(), storage.ReportStatus_PREPARING.String()).ProtoQuery()
+	reportSnapshots, err := s.snapshotDatastore.SearchReportSnapshots(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	for _, reportSnapshot := range reportSnapshots {
+		if currentConfig.Creator.GetId() == reportSnapshot.GetRequester().GetId() {
```

Comment:
We don't want to compare the config creator == report requester here. What we want to compare is if the user requesting update on the config == report requester. 
You get the user requesting update from the context. See example on how to get the user form context in `CancelReport` or `DownloadReport` implementations.